File: backend/app/domain/models/tourist_profile.py
import uuid
from typing import TYPE_CHECKING, Optional
from sqlalchemy import String, Text, Boolean, ForeignKey, Uuid
from sqlalchemy.orm import Mapped, mapped_column, relationship
from backend.app.domain.models.base import UUIDModel

if TYPE_CHECKING:
    from backend.app.domain.models.user import User


class TouristProfile(UUIDModel):
    __tablename__ = "tourist_profiles"

    user_id: Mapped[uuid.UUID] = mapped_column(
        Uuid(as_uuid=True),
        ForeignKey("users.id", ondelete="CASCADE"),
        unique=True,
        nullable=False,
        index=True,
    )
    nationality: Mapped[Optional[str]] = mapped_column(
        String(100),
        nullable=True,
    )
    passport_or_id_hash: Mapped[Optional[str]] = mapped_column(
        String(255),
        nullable=True,
    )
    emergency_contact_name: Mapped[Optional[str]] = mapped_column(
        String(255),
        nullable=True,
    )
    emergency_contact_phone: Mapped[Optional[str]] = mapped_column(
        String(50),
        nullable=True,
    )
    medical_notes: Mapped[Optional[str]] = mapped_column(
        Text,
        nullable=True,
    )
    consent_given: Mapped[bool] = mapped_column(
        Boolean,
        default=False,
        nullable=False,
    )

    # Relationships
    user: Mapped["User"] = relationship(
        "User",
        back_populates="tourist_profile",
    )

File: apps/mobile/lib/domain/models/tourist_profile.dart
class TouristProfileModel {
  final String id;
  final String userId;
  final String? nationality;
  final String? emergencyContactName;
  final String? emergencyContactPhone;
  final String? medicalNotes;
  final bool consentGiven;

  TouristProfileModel({
    required this.id,
    required this.userId,
    this.nationality,
    this.emergencyContactName,
    this.emergencyContactPhone,
    this.medicalNotes,
    required this.consentGiven,
  });

  factory TouristProfileModel.fromJson(Map<String, dynamic> json) {
    return TouristProfileModel(
      id: json['id'] as String,
      userId: json['user_id'] as String,
      nationality: json['nationality'] as String?,
      emergencyContactName: json['emergency_contact_name'] as String?,
      emergencyContactPhone: json['emergency_contact_phone'] as String?,
      medicalNotes: json['medical_notes'] as String?,
      consentGiven: json['consent_given'] as bool? ?? false,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'nationality': nationality,
      'emergency_contact_name': emergencyContactName,
      'emergency_contact_phone': emergencyContactPhone,
      'medical_notes': medicalNotes,
      'consent_given': consentGiven,
    };
  }
}
